public/app/themes/justice/inc/search.php

```diff
@@ -319,12 +314,77 @@ public function removeColumns(array $columns): array
      *
      * @return void
      */
-
     public function removeSearchesSubMenus()
     {
         if (!current_user_can('manage_options')) {
             remove_submenu_page('index.php', 'relevanssi-premium/relevanssi.php');
             remove_submenu_page('index.php', 'relevanssi_admin_search');
         }
     }
+
+    /**
+     * Handle malformed search URLs where the path has multiple pages.
+     *
+     * e.g /search/the/page/page/11
+     *
+     * @return void
+     */
+    public function redirectMultiplePageInURI(): void
+    {
+        // Trim the first and last slash
+        $uri = trim($_SERVER['REQUEST_URI'], '/');
+        // Split the URI by '/'
+        $uri_parts = explode('/', $uri);
+
+        // Check if the URI has at least 4 parts and the first part is 'search'
+        if (sizeof($uri_parts) < 4 || $uri_parts[0] !== 'search') {
+            return;
+        }
+
+        // Remove the first 2 from the array
+        $uri_parts = array_slice($uri_parts, 2);
+
+        // Count the number of times 'page' appears in the $uri_parts
+        $pages_count = array_count_values($uri_parts)['page'];
+
+        if ($pages_count > 1) {
+            // Redirect to the search page
+            $url = home_url('/search');
+            wp_redirect($url);
+            exit;
+        }
+    }
+
+
+    /**
+     * Handle malformed search URLs with arrays in the query string.
+     *
+     * This function will redirect the user to the search page if the query string contains arrays.
+     * e.g. /search?audience[$testing]=1 or /search?audience%5B%24testing%5D=1
+     *
```

Comment:
@EarthlingDavey what's causing this to happen? i.e. what's the flow for a user to end up with an array in the query string or multiple instances of 'page' in the url?

Reply:
I think it's a vuln. scanner running fuzzing to see what will cause errors.

I don't think we can protect against all fuzzing attacks - some will cause warnings and others will cause errors that will show a generic 500 error page.

But, this one in particular was causing the search plugin to error, and seemed like something that should be avoided. Especially as it has a simple-ish solution that would not get in the way of other site functionality.

Reply:
@EarthlingDavey that makes sense, thanks for clarifying!

Reply:
No probs :) Thanks for the review